src/js/components/DataTable/DataTable.js

```diff
@@ -160,20 +160,20 @@ const DataTable = ({
 
   const onHeaderWidths = useCallback(
     (columnWidths) => {
-      const pinnedProperties = columns
+      const hasSelectColumn = Boolean(select || onSelect);
+      let pinnedProperties = columns
         .map((pinnedColumn) => pinnedColumn.pin && pinnedColumn.property)
         .filter((n) => n);
-
+      if (hasSelectColumn && pinnedProperties.length > 0) {
+        pinnedProperties = ['_select', ...pinnedProperties];
```

Comment:
I'd love to find a cleaner solution for the checkbox column than giving it the internal name `_select` if anyone has an idea.